src/System.Device.Gpio/System/Device/ComponentInformation.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+#pragma warning disable CS1591
+namespace System.Device
+{
+    public record ComponentInformation
+    {
+        private readonly List<ComponentInformation> _subComponents;
+
+        public ComponentInformation(object instance, string name)
+            : this(instance, name, string.Empty)
```

Comment:
This is a record, why not using the primary ctor and optional parameters with a default value?

Reply:
I prefer explicit overloads, but that's just syntax and can be adjustedat the end.